### PR TITLE
project export coordinator

### PR DESCRIPTION
**Pull request by jjidt**
write md files, download assets, and zip it up

TODO:

- .env temp file path
- controller perms
- worker/upload to s3

### CODE REVIEW

File: api/src/integrators/zip.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  })
+}
+
+const getRecursiveDirectoryFiles = async function(dir) {
+  const entries = await fs.promises.readdir(dir, { withFileTypes: true })
+  const files = await Promise.all(entries.map((entry) => {
+    const entryPath = path.resolve(dir, entry.name)
+    return entry.isDirectory() ? getRecursiveDirectoryFiles(entryPath) : entryPath
+  }))
+  return Array.prototype.concat(...files)
```
Comment (dheniges) [inline api/src/integrators/zip.js:68]: This is returning a new array of the files, which are arrays or strings it looks like. But why the new array vs returning `files` here?

**Review by dheniges — CHANGES_REQUESTED** [review]
One code question (see comment), and then:
- if I have a date field, with or without a value, I get an error: cannot read property `toISOString` of null
- A project with 1 document with a single text field just hangs when I try to export it.

I'm running the export through postman

File: api/src/integrators/zip.js (hunk 1 above)
Comment (jjidt) [reply api/src/integrators/zip.js:68]: just a shortcut to flatten a multi dimensional array, which files ends up being

**Comment by dheniges** [comment]
This work is now in the feature branch via another PR, closing

**Pull request closed without merging**